diff --git a/clutter/clutter-stage.c b/clutter/clutter-stage.c
index 2c67ca3..d281672 100644
--- a/clutter/clutter-stage.c
+++ b/clutter/clutter-stage.c
@@ -3795,8 +3795,8 @@ _clutter_stage_schedule_update (ClutterStage *stage)
   if (stage_window == NULL)
     return;
 
-  return _clutter_stage_window_schedule_update (stage_window,
-                                                stage->priv->sync_delay);
+  _clutter_stage_window_schedule_update (stage_window,
+                                         stage->priv->sync_delay);
 }
 
 /* Returns the earliest time the stage is ready to update */
