diff -bruN a/gtk/gtkmarshal.h b/gtk/gtkmarshal.h
--- a/gtk/gtkmarshal.h	2018-01-09 05:12:03.000000000 +0900
+++ b/gtk/gtkmarshal.h	2020-04-10 11:53:21.641118200 +0900
@@ -101,17 +101,17 @@
                                          gpointer      marshal_data);
 
 /* NONE:BOOL (./gtkmarshal.list:10) */
-#define gtk_marshal_VOID__BOOLEAN	g_cclosure_marshal_VOID__BOOLEAN
+/* #define gtk_marshal_VOID__BOOLEAN	g_cclosure_marshal_VOID__BOOLEAN */
 
 #define gtk_marshal_NONE__BOOL	gtk_marshal_VOID__BOOLEAN
 
 /* NONE:BOXED (./gtkmarshal.list:11) */
-#define gtk_marshal_VOID__BOXED	g_cclosure_marshal_VOID__BOXED
+/* #define gtk_marshal_VOID__BOXED	g_cclosure_marshal_VOID__BOXED */
 
 #define gtk_marshal_NONE__BOXED	gtk_marshal_VOID__BOXED
 
 /* NONE:ENUM (./gtkmarshal.list:12) */
-#define gtk_marshal_VOID__ENUM	g_cclosure_marshal_VOID__ENUM
+/* #define gtk_marshal_VOID__ENUM	g_cclosure_marshal_VOID__ENUM */
 
 #define gtk_marshal_NONE__ENUM	gtk_marshal_VOID__ENUM
 
@@ -138,7 +138,7 @@
 #define gtk_marshal_NONE__ENUM_FLOAT_BOOL	gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN
 
 /* NONE:INT (./gtkmarshal.list:15) */
-#define gtk_marshal_VOID__INT	g_cclosure_marshal_VOID__INT
+/* #define gtk_marshal_VOID__INT	g_cclosure_marshal_VOID__INT */
 
 #define gtk_marshal_NONE__INT	gtk_marshal_VOID__INT
 
@@ -165,17 +165,17 @@
 #define gtk_marshal_NONE__INT_INT_POINTER	gtk_marshal_VOID__INT_INT_POINTER
 
 /* NONE:NONE (./gtkmarshal.list:18) */
-#define gtk_marshal_VOID__VOID	g_cclosure_marshal_VOID__VOID
+/* #define gtk_marshal_VOID__VOID	g_cclosure_marshal_VOID__VOID */
 
 #define gtk_marshal_NONE__NONE	gtk_marshal_VOID__VOID
 
 /* NONE:OBJECT (./gtkmarshal.list:19) */
-#define gtk_marshal_VOID__OBJECT	g_cclosure_marshal_VOID__OBJECT
+/* #define gtk_marshal_VOID__OBJECT	g_cclosure_marshal_VOID__OBJECT */
 
 #define gtk_marshal_NONE__OBJECT	gtk_marshal_VOID__OBJECT
 
 /* NONE:POINTER (./gtkmarshal.list:20) */
-#define gtk_marshal_VOID__POINTER	g_cclosure_marshal_VOID__POINTER
+/* #define gtk_marshal_VOID__POINTER	g_cclosure_marshal_VOID__POINTER */
 
 #define gtk_marshal_NONE__POINTER	gtk_marshal_VOID__POINTER
 
@@ -279,7 +279,7 @@
 #define gtk_marshal_NONE__POINTER_UINT_UINT	gtk_marshal_VOID__POINTER_UINT_UINT
 
 /* NONE:STRING (./gtkmarshal.list:31) */
-#define gtk_marshal_VOID__STRING	g_cclosure_marshal_VOID__STRING
+/* #define gtk_marshal_VOID__STRING	g_cclosure_marshal_VOID__STRING */
 
 #define gtk_marshal_NONE__STRING	gtk_marshal_VOID__STRING
 
@@ -295,7 +295,7 @@
 #define gtk_marshal_NONE__STRING_INT_POINTER	gtk_marshal_VOID__STRING_INT_POINTER
 
 /* NONE:UINT (./gtkmarshal.list:33) */
-#define gtk_marshal_VOID__UINT	g_cclosure_marshal_VOID__UINT
+/* #define gtk_marshal_VOID__UINT	g_cclosure_marshal_VOID__UINT */
 
 #define gtk_marshal_NONE__UINT	gtk_marshal_VOID__UINT
 
